File: client-application/src/BookService/BookService.js
import axios from "axios";

export class BookService {
    static async getAllBooksAsync(searchString){
        if (searchString === null){
            const response = await axios.get(`https://localhost:7273/api/books`);
            return response;
        }
        
        const response = await axios.get(`https://localhost:7273/api/books?searchString=${searchString}`);
        return response;
    }

    static async createBookAsync(title) {
        await axios.post(`https://localhost:7273/api/books`, title);
    }

    static async deleteBookAsync(id) {
        await axios.delete(`https://localhost:7273/api/books/${id}`);
    }
}

File: Application/Services/BookService.cs
using Application.Interfaces;
using CSharpFunctionalExtensions;
using Domain.Entities;
using Infrastructure.EntityFrameworkCore.Interfaces;

namespace Application.Services
{
    public class BookService(IBookRepository bookRepository) : IBookService
    {
        private readonly IBookRepository _bookRepository = bookRepository;

        public async Task<Result<List<Book>>> GetAllBooksAsync(string? searchString, CancellationToken ct)
        {
            return await _bookRepository.GetAllBooksAsync(searchString, ct);
        }

        public async Task<Result<Book>> CreateBookAsync(string title, CancellationToken ct)
        {
            Result<Book> resultBook = Book.Create(
                Guid.NewGuid(),
                title);

            if (resultBook.IsFailure)
                return Result.Failure<Book>(resultBook.Error);

            return await _bookRepository.CreateBookAsync(resultBook.Value, ct);
        }

        public async Task<Result<Guid>> DeleteBookAsync(Guid id, CancellationToken ct)
        {
            return await _bookRepository.DeleteBookAsync(id, ct);
        }
    }
}
